tls/s2n_connection.h

```diff
@@ -201,7 +201,9 @@ struct s2n_connection {
     /* Our crypto parameters */
     struct s2n_crypto_parameters initial;
     struct s2n_crypto_parameters secure;
-    struct s2n_secrets secrets;
+    union s2n_secrets secrets;
+    uint8_t client_random[S2N_TLS_RANDOM_DATA_LEN];
+    uint8_t server_random[S2N_TLS_RANDOM_DATA_LEN];
```

Comment:
I do think this is weird to put these two arrays on the connection. Why do they not belong in the handshake_params struct? Aren't they basically handshake parameters? Or actually the kex_parameters struct? I think both of those make more sense then the connection.

Reply:
I discuss this briefly in the description.
* Putting them in the "kex_parameters" struct feels wrong because they're only used for key exchange in TLS1.2; in TLS1.3 they're used for downgrade protection and middlebox compatibility related to hello retries. It ends up looking kind of funny when we're handling kex_params.server_random in those cases.
* Putting them in the "handshake_params" feels wrong because that's currently mostly fields related to authentication. But the name / comment doesn't require that, so maybe that's the right place.